pkg/api/endpoints/util.go

```diff
@@ -64,12 +65,37 @@ func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 				}
 				portsToAddrs[*epp].Insert(allAddrs[ak])
 			}
+			for k := range subsets[i].NotReadyAddresses {
```

Comment:
Removed duplication, also added comment that NotReady trumps Ready if we
see the same address twice.

On Fri, Sep 11, 2015 at 2:18 PM, Derek Carr <notifications@github.com>
wrote:

> In pkg/api/endpoints/util.go
> <https://github.com/kubernetes/kubernetes/pull/13778#discussion_r39300442>
> :
>
> > @@ -64,12 +65,37 @@ func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
> >  				}
> >  				portsToAddrs[*epp].Insert(allAddrs[ak])
> >  			}
> > +			for k := range subsets[i].NotReadyAddresses {
>
> it feels like there is a lot of code duplication here with what is in
> subsets[i].Addresses, would be good if we could somehow make something
> common.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13778/files#r39300442>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift